### PR TITLE
Improve spelling in source files

### PR DESCRIPTION
**Pull request by gustafn**
see ticket at https://jira.mongodb.org/browse/CDRIVER-2758

### CODE REVIEW

File: src/libbson/src/bson/bson.h [hunk 1]
```diff
@@ -936,7 +936,7 @@ bson_append_oid (bson_t *bson,
  *   's' for dotall mode ('.' matches everything)
  *   'u' to make \w and \W match unicode.
  *
- * For more information on what comprimises a BSON regex, see bsonspec.org.
+ * For more information on what compromises a BSON regex, see bsonspec.org.
```
Comment (ajdavis) [inline src/libbson/src/bson/bson.h]: comprises, here and below

File: src/libbson/doc/bson_array_as_json.rst [hunk 2]
```diff
@@ -22,7 +22,7 @@ Description
 
 The :symbol:`bson_array_as_json()` function shall encode ``bson`` as a UTF-8
 string using libbson's legacy JSON format, except the outermost element is
-encoded as JSON array, rather then JSON document.
+encoded as JSON array, rather than JSON document.
```
Comment (ajdavis) [inline src/libbson/doc/bson_array_as_json.rst]: "as a JSON array, rather than a JSON document"

**Comment by rcsanchez97** [comment]
On Thu, Jul 19, 2018 at 12:24:56PM -0700, A. Jesse Jiryu Davis wrote:
>  - * For more information on what comprimises a BSON regex, see bsonspec.org.
>  + * For more information on what compromises a BSON regex, see bsonspec.org.
> 
>    comprises, here and below
> 
This would be a misuse of 'comprises'.  A better here word would be
'constitutes'.

**Comment by ajdavis** [comment]
Good point, let's use "constitutes".

File: src/libbson/doc/bson_array_as_json.rst (hunk 2 above)
Comment (gustafn) [reply src/libbson/doc/bson_array_as_json.rst]: adding the "as" is certainly better

Comment (ajdavis) [reply src/libbson/doc/bson_array_as_json.rst]: OK!

**Comment by gustafn** [comment]
or maybe "For detailed information about BSON regex, see bsonspec.org."

**Comment by ajdavis** [comment]
Sure, even better. =)

**Comment by ajdavis** [comment]
Merged in https://github.com/mongodb/mongo-c-driver/commit/62c9c3ca7fef8601f486fa0c3b46f459aa7aae7b, thank you!

**Pull request closed without merging**